src/Beacon.jsx

```diff
@@ -11,7 +11,20 @@ const TOOLTIP_OVERLAY_CLASS = 'tour-overlay';
 const TOOLTIP_FADED_CLASS = 'tour-faded';
 const TARGET_CLONE_ID = 'tour-target-clone';
 // If margins are within tolerance then we center tooltip
-const TOOLTIP_TOLERANCE = 50;
+// Tolerance is ratio of margins to total dimension (height or width)
+// So if the difference between the tooltip margins is less than
+// `dimension/4` then we center it.
+const TOOLTIP_TOLERANCE_RATIO = 4;
+
+const TOOLTIP_STATE_VARIABLES = [
+  'tooltipActive',
+  'tooltipHeight',
+  'tooltipWidth',
+  'tooltipHidden',
+  'tooltipAttachment',
+  'appRoot',
+  'appRootClassName'
+];
```

Comment:
As discussed the maintenance of two separate states (for `tooltip` equals `true` and `false`) along with `shouldComponentUpdate` is an anti-pattern. In a future version I'll move the two different "modes" into separate subcomponents with their own state.